### PR TITLE
Feature/paginate

### PR DESCRIPTION
**Pull request by dwoznicka**

### CODE REVIEW

File: client/css/pagination.scss [hunk 1]
```diff
@@ -0,0 +1,38 @@
+.paginate-nav {
```
Comment (blimusiek) [inline client/css/pagination.scss]: @dwoznicka could you consider moving colors and paddings values to separate variable ?

File: client/js/actions/snippet-actions.js [hunk 2]
```diff
@@ -10,26 +10,17 @@ class SnippetActions {
...
-    axios.get('/snippets')
-      .then(res => {
-        FlashMessages.pushMessage({ content: 'Here are all of the snippets!' });
-        this.dispatch({ok: true, snippets: res.data});
-      }).catch(err => {
-        FlashMessages.pushMessage({ content: 'Oops! Something went wrong :(' });
-        this.dispatch({ok: false, error: err});
-      });
+  getPaginatedSnippets(page, perPage){
+    var start = (perPage*(page-1));
```
Comment (blimusiek) [inline client/js/actions/snippet-actions.js]: @dwoznicka could you plese change variable `start` name to `offset` ?

File: client/js/actions/user-snippets-actions.js [hunk 3]
```diff
@@ -5,15 +5,17 @@ import FlashMessages from './flash-messages-actions';
...
-  getAllOfCurrentUser() {
-    axios.get('/snippets/user')
-      .then(res => {
-        FlashMessages.pushMessage({ content: 'Here are your own snippets!' });
-        this.dispatch({ok: true, snippets: res.data});
-      }).catch(err => {
-        FlashMessages.pushMessage({ content: 'Oops! Something went wrong :(' });
-        this.dispatch({ok: false, error: err});
-      });
+  getPaginatedUserSnippets(page, perPage){
```
Comment (blimusiek) [inline client/js/actions/user-snippets-actions.js]: @dwoznicka you are repeating yourself consider moving that offset counting and url bulding to some util pagination lib and importing it and using it.

File: client/js/components/pages/paginator.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    super(props);
+  }
+
+  _handleClick(page, e){
+    e.preventDefault();
+    this.props.onClickPage(page);
+  }
+
+  render() {
+    let p = this.props;
```
Comment (blimusiek) [inline client/js/components/pages/paginator.jsx]: @dwoznicka please be consistent and use `p` or `this.props` :)

File: client/js/components/pages/paginator.jsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        </li>
+      );
+    });
+
+
+    return (
+      <nav className={'paginate-nav'}>
+        <ul className={'pagination'}>
+          <li>
+            <a onClick={this._handleClick.bind(this, p.currentPage-1)}> &laquo; Prev</a>
```
Comment (blimusiek) [inline client/js/components/pages/paginator.jsx]: @dwoznicka please do not display that link if user is on the first page

File: client/js/components/pages/snippets-index.jsx [hunk 6]
```diff
@@ -13,13 +14,38 @@ export default class SnippetsIndex extends React.Component {
...
   }
 
   componentDidMount() {
     this.storeListeners = [];
     this.storeListeners.push(SnippetStore.listen(this._onChange));
     this.storeListeners.push(SnippetSearchStore.listen(this._onSearch));
-    SnippetActions.getAll();
+    this._getPaginatedSnippets(1);
+    SnippetActions.getCount();
+    this.setState({
```
Comment (blimusiek) [inline client/js/components/pages/snippets-index.jsx]: @dwoznicka I think it is not necessary here

File: client/js/components/pages/snippets-index.jsx [hunk 7]
```diff
@@ -13,13 +14,38 @@ export default class SnippetsIndex extends React.Component {
...
+    });
+  }
+
+  _goToPage(page){
+    var perPage = this.props.route.perPage;
+    var allPages = Math.ceil(this.state.totalCount/perPage);
+
+    if(page>0 && page <= allPages){
+      this._getPaginatedSnippets(page);
+      this.setState({
```
Comment (blimusiek) [inline client/js/components/pages/snippets-index.jsx]: @dwoznicka it can be removed

File: client/js/components/pages/user-snippets.jsx [hunk 8]
```diff
@@ -3,18 +3,24 @@ import PageWrapper from '../page-wrapper';
...
+    this._goToPage = this._goToPage.bind(this);
   }
 
   componentDidMount() {
     this.storeListeners = [];
     this.storeListeners.push(UserSnippetsStore.listen(this._onChange));
-    UserSnippetsActions.getAllOfCurrentUser();
+    this._getPaginatedSnippets(1);
+    UserSnippetsActions.getCount();
+    this.setState({
```
Comment (blimusiek) [inline client/js/components/pages/user-snippets.jsx]: @dwoznicka set is already set to proper currentPage

File: client/js/actions/snippet-actions.js [hunk 9]
```diff
@@ -10,26 +10,17 @@ class SnippetActions {
...
-    axios.get('/snippets/user')
-      .then(res => {
-        FlashMessages.pushMessage({ content: 'Here are your own snippets!' });
-        this.dispatch({ok: true, snippets: res.data});
-      }).catch(err => {
-        FlashMessages.pushMessage({ content: 'Oops! Something went wrong :(' });
-        this.dispatch({ok: false, error: err});
-      });
+  getCount(){
+    axios.get('/snippets/count')
```
Comment (vrael560) [inline client/js/actions/snippet-actions.js]: why separate action for `count`?

**Pull request merged**